lib/redis/connection/memory.rb

```diff
@@ -955,6 +955,38 @@ def zrange(key, start, stop, with_scores = nil)
         results[start..stop].flatten.map(&:to_s)
       end
 
+      def zrangebylex(key, start, stop, *opts)
+        data_type_check(key, ZSet)
+        return [] unless data[key]
+        zset = data[key]
+
+        sorted = if zset.identical_scores?
+          zset.keys.sort { |x, y| x.to_s <=> y.to_s }
+        else
+          zset.keys
+        end
+
+        range = get_range start, stop, sorted.first, sorted.last
+
+        filtered = []
+        sorted.each do |element|
```

Comment:
instead of traversing the `sorted` array, maybe you could use `bsearch` for searching the beginning and ending positions.